components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```
@@ -0,0 +1,370 @@
+                        handleErrorResponse((UMAException) throwable, true);
+                        log.error("Internal server error occurred. ", throwable);
+                    }
+                } else {
+                    throw new UMAClientException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID);
+                }
+            } catch (UMAClientException e) {
+                handleErrorResponse(e, false);
+                log.error("Client error when retrieving resource from the resource server.", e);
+            }
+            return response;
+        } else {
+            response = Response.status(Response.Status.UNAUTHORIZED).build();
+        }
+        return response;
+    }
+
+    /**
+     * Delete the resource for the given resourceId
+     *
+     * @param resourceId resourceId of the resource which need to get deleted
+     * @return Response with the status of resource deletion
+     */
+    @Override
+    public Response deleteResource(String resourceId, MessageContext context) {
+
+        Response response = null;
+        if (!isValidateToken(context)) {
+            try {
+                if (isResourceIdValid(resourceId)) {
+
+                    try {
+                        if (ResourceUtils.getResourceService().deleteResource(resourceId)) {
+                            return Response.status(Response.Status.NO_CONTENT).build();
+                        }
+
+                    } catch (UMAServiceException e) {
+                        handleErrorResponse(e, true);
+                        log.error("Invalid request. ", e);
+
+                    } catch (Throwable throwable) {
+                        handleErrorResponse((UMAException) throwable, true);
+                        log.error("Internal server error occurred. ", throwable);
+                    }
+                } else {
+                    throw new UMAClientException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID);
```

Shall we return a error code